verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -81,13 +83,38 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
     }
     const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if (exceptions_.size()) {
+      name_filtered.reserve(name.length());
+      const auto& name_split = absl::StrSplit(name, '_');
+      for (const auto& ns : name_split) {
+        if (std::find(exceptions_.begin(), exceptions_.end(), ns) ==
+            exceptions_.end()) {
+          name_filtered.append(ns);
+        }
+      }
+    }
+    if (!verible::IsLowerSnakeCaseWithDigits(
+            name_filtered.length() ? name_filtered : name) ||
```

Comment:
I'd probably put this in a named variable before used here
```
const absl::string_view relevant_name = name_filtered.empty() ? name : name_filtered;
```

A question: what if the name _only_ contains exceptioned name parts ? In that case, we end up with an empty string, so we choose to use the `name` instead ...
So we can't really use the name length as critterium here; Maybe `exceptions_.empty() ? name : name_filtered;`

But then what happens if we pass an empty string to `IsLowerSnakeCaseWithDigits()` ?

To start, I'd add a test to figure out what happens in that case. Then fix it here.